packages/teleport/src/components/BannerList/BannerList.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useEffect, useState } from 'react';
+
+import { Box } from 'design';
+
+import { Banner } from './Banner';
+
+export const BannerList = ({ banners = [] }: Props) => {
+  const [bannerList, setBannerList] = useState<{ [id: string]: BannerType }>(
+    {}
+  );
+
+  useEffect(() => {
+    const newList = {};
+    banners.forEach(banner => (newList[banner.id] = banner));
+    Object.assign(newList, bannerList);
+    setBannerList(newList);
+  }, [banners]);
+
+  const removeBanner = id => {
+    const newList = { ...bannerList };
+    newList[id].hidden = true;
```

Comment:
this still mutates the original bannerList[id] object reference. You need to shallow copy both the root object and target object 
```
const newList = { ...bannerList, [id]: { ...bannerList[id], hidden: true } };
```

Reply:
Yeah I wasn't concerned about modifying the nested object as I only needed the pointer of the container to change to trigger a re-render.

Reply:
Right good catch, I applied your suggestion above but also properly cloned the `banners` prop as was my original intention.